### PR TITLE
Authentication(Token) Login for User and Host

### PR DESCRIPTION
**Pull request by dlamli**
- LoginController Removed.
- User and Host Login with AllowAnonymous with Token.
- The login will be with Username(Name) and Password(Password).
Notes:
Postman collection Updated:
[Ulacitbnb.postman_collection.zip](https://github.com/DLam28/ulacitbnb/files/6759129/Ulacitbnb.postman_collection.zip)

### CODE REVIEW

File: Controllers/HostController.cs [hunk 1]
```diff
@@ -139,6 +139,63 @@ public HttpResponseMessage CreateNewHost(Host host)
             }
             return Request.CreateResponse(HttpStatusCode.OK, $"NEW HOST CREATED SUCCESFULLY: {host.Name} {host.LastName}");
         }
+        // ===================================================================================================
+        [AllowAnonymous]
+        [HttpPost]
+        [Route("auth")]
```
Comment (Gabuardi) [inline Controllers/HostController.cs]: Methods attributes can be declared on this way which looks cleaner

```suggestion
 [HttpPost, AllowAnonymous, Route("auth")]
```

File: Controllers/HostController.cs [hunk 2]
```diff
@@ -139,6 +139,63 @@ public HttpResponseMessage CreateNewHost(Host host)
...
+                        host.Name = sqlDataReader.GetString(1);
+                        host.LastName = sqlDataReader.GetString(2);
+                        host.Password = sqlDataReader.GetString(3);
+                        host.Description = sqlDataReader.GetString(4);
+                        host.Status = sqlDataReader.GetString(5);
+
+                        var token = TokenGenerator.GenerateTokenJwt(loginRequest.Username);
+                        host.Token = token;
+                    }
+                    sqlConnection.Close();
```
Comment (Gabuardi) [inline Controllers/HostController.cs]: when a SQL connection is open inside a `using()` the connection is closed when the execution go out from the `using()` so is not necessary call a `Close()` inside a `using()`

Reference: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement

File: Controllers/HostController.cs [hunk 3]
```diff
@@ -139,6 +139,63 @@ public HttpResponseMessage CreateNewHost(Host host)
...
+
+                    if (!string.IsNullOrEmpty(host.Token))
+                        return Ok(host);
+                    else return Unauthorized();
+                }
+            }
+            catch (Exception ex)
+            {
+
```
Comment (Gabuardi) [inline Controllers/HostController.cs:187]: empty space unecessary

File: Controllers/UserController.cs [hunk 4]
```diff
@@ -109,6 +108,67 @@ public IHttpActionResult GetAll()
             return Ok(users);
         }
         //------------------------------------------------------------------------------
+        [AllowAnonymous]
+        [HttpPost]
+        [Route("auth")]
```
Comment (Gabuardi) [inline Controllers/UserController.cs]: Methods attributes can be declared on this way which looks cleaner

```suggestion
 [HttpPost, AllowAnonymous, Route("auth")]
```

File: Controllers/UserController.cs [hunk 5]
```diff
@@ -109,6 +108,67 @@ public IHttpActionResult GetAll()
...
+                        user.Use_Password = sqlDataReader.GetString(4);
+                        user.Use_Email = sqlDataReader.GetString(5);
+                        user.Use_Status = sqlDataReader.GetString(6);
+                        user.Use_BirthDate = sqlDataReader.GetDateTime(7);
+                        user.Use_Phone = sqlDataReader.GetString(8);
+
+                        var token = TokenGenerator.GenerateTokenJwt(loginRequest.Username);
+                        user.Token = token;
+                    }
+                    sqlConnection.Close();
```
Comment (Gabuardi) [inline Controllers/UserController.cs]: when a SQL connection is open inside a using() the connection is closed when the execution go out from the using() so is not necessary call a Close() inside a using()

Reference: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement

File: Controllers/UserController.cs [hunk 6]
```diff
@@ -109,6 +108,67 @@ public IHttpActionResult GetAll()
...
+
+                    if (!string.IsNullOrEmpty(user.Token))
+                        return Ok(user);
+                    else return Unauthorized();
+                }
+            }
+            catch (Exception ex)
+            {
+
```
Comment (Gabuardi) [inline Controllers/UserController.cs]: empty space unecessary

**Review by Gabuardi — CHANGES_REQUESTED** [review]
Also please take in consideration the new logic also require a different registration for user and host, so the method `create` must to be public in both controllers.

File: Controllers/HostController.cs (hunk 1 above)
Comment (dlamli) [reply Controllers/HostController.cs]: Allright, thanks for the feedback!

File: Controllers/HostController.cs (hunk 2 above)
Comment (Gabuardi) [reply Controllers/HostController.cs]: Still there

**Review by Gabuardi — APPROVED** [review]

**Pull request merged**